File: app/src/main/java/com/codeart/tangdi/catfat/netrequest/VolleyRequest.java
package com.codeart.tangdi.catfat.netrequest;

import android.graphics.Bitmap;
import android.widget.ImageView;

import com.android.volley.AuthFailureError;
import com.android.volley.Request;
import com.android.volley.Response;
import com.android.volley.toolbox.ImageRequest;
import com.android.volley.toolbox.JsonObjectRequest;

import org.json.JSONObject;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

/**
 * Created by Administrator on 2017/6/2.
 */

public class VolleyRequest {

    public ImageRequest LoadImage(String url, Response.Listener<Bitmap> listener, Response.ErrorListener errorListener){
        ImageRequest imageRequest = new ImageRequest(url, listener, 0, 0, ImageView.ScaleType.FIT_XY,
                                                     Bitmap.Config.ARGB_8888, errorListener);
        return imageRequest;
    }

    public JsonObjectRequest JSONRequestGet(String url, JSONObject jsonObject,
                                            Response.Listener<JSONObject> listener,
                                            Response.ErrorListener errorListener){
        return new JsonObjectRequest(Request.Method.GET, url, null, listener, errorListener);
    }

    public StringJsonRequest StringJSONRequestGet(String url, Response.Listener<JSONObject> listener,
                                            Response.ErrorListener errorListener, Map<String, String> map){
        StringBuilder stringBuilder = new StringBuilder(url);
        stringBuilder.append(getParam(map));
        return new StringJsonRequest(stringBuilder.toString(), listener, errorListener);
    }

    public StringJsonRequest StringJSONRequestPost(String url,
                                                   Response.Listener<JSONObject> listener,
                                                   Response.ErrorListener errorListener, Map<String, String> request){
        return new StringJsonRequest(Request.Method.POST, url, listener, errorListener, request);

    }

        private String getParam(Map<String, String> map){
            if(map == null){
                return "";
            }
            StringBuilder stringBuilder = new StringBuilder();
            stringBuilder.append("?");
            for (Map.Entry<String, String> entry : map.entrySet()) {
                stringBuilder.append(entry.getKey());
                stringBuilder.append("=");
                stringBuilder.append(entry.getValue());
                stringBuilder.append("&");
            }
            return stringBuilder.substring(0, stringBuilder.length() - 1);
    }

    public StringBinaryRequest StringBinaryRequestPost(String url, Response.Listener<byte[]> listener,
                                                      Response.ErrorListener errorListener,
                                                      Map<String, String> requset){
        return new StringBinaryRequest(url, listener, errorListener, requset);
    }

    public StringBinaryRequest StringBinayRequestGet(String url, Response.Listener<byte[]> listener,
                                                     Response.ErrorListener errorListener,
                                                     Map<String, String> requset){
        StringBuilder stringBuilder = new StringBuilder(url);
        stringBuilder.append(getParam(requset));
        return new StringBinaryRequest(url, listener, errorListener);

    }



}

File: app/src/main/java/com/codeart/tangdi/catfat/netrequest/VolleyRequest.kt
package com.codeart.tangdi.catfat.netrequest

import android.graphics.Bitmap
import android.widget.ImageView
import com.android.volley.Request
import com.android.volley.Response
import com.android.volley.toolbox.ImageRequest
import com.android.volley.toolbox.JsonObjectRequest
import org.json.JSONObject

/**
 * Created by Administrator on 2017/6/2.
 */

class VolleyRequest {

    fun LoadImage(url: String, listener: Response.Listener<Bitmap>, errorListener: Response.ErrorListener): ImageRequest {
        val imageRequest = ImageRequest(url, listener, 0, 0, ImageView.ScaleType.FIT_XY,
                Bitmap.Config.ARGB_8888, errorListener)
        return imageRequest
    }

    fun JSONRequestGet(url: String, jsonObject: JSONObject,
                       listener: Response.Listener<JSONObject>,
                       errorListener: Response.ErrorListener): JsonObjectRequest {
        return JsonObjectRequest(Request.Method.GET, url, null, listener, errorListener)
    }

    fun StringJSONRequestGet(url: String, listener: Response.Listener<JSONObject>,
                             errorListener: Response.ErrorListener, map: Map<String, String>): StringJsonRequest {
        val stringBuilder = StringBuilder(url)
        stringBuilder.append(getParam(map))
        return StringJsonRequest(stringBuilder.toString(), listener, errorListener)
    }

    fun StringJSONRequestPost(url: String,
                              listener: Response.Listener<JSONObject>,
                              errorListener: Response.ErrorListener, request: Map<String, String>): StringJsonRequest {
        return StringJsonRequest(Request.Method.POST, url, listener, errorListener, request)

    }

    private fun getParam(map: Map<String, String>?): String {
        if (map == null) {
            return ""
        }
        val stringBuilder = StringBuilder()
        stringBuilder.append("?")
        for ((key, value) in map) {
            stringBuilder.append(key)
            stringBuilder.append("=")
            stringBuilder.append(value)
            stringBuilder.append("&")
        }
        return stringBuilder.substring(0, stringBuilder.length - 1)
    }

    fun StringBinaryRequestPost(url: String, listener: Response.Listener<ByteArray>,
                                errorListener: Response.ErrorListener,
                                requset: Map<String, String>): StringBinaryRequest {
        return StringBinaryRequest(url, listener, errorListener, requset)
    }

    fun StringBinayRequestGet(url: String, listener: Response.Listener<ByteArray>,
                              errorListener: Response.ErrorListener,
                              requset: Map<String, String>?): StringBinaryRequest {
        val stringBuilder = StringBuilder(url)
        stringBuilder.append(getParam(requset))
        return StringBinaryRequest(url, listener, errorListener)

    }


}
